test/utils/test_utils.py

```diff
@@ -65,117 +65,93 @@ def test_validate_controls():
 
 
 @pytest.mark.parametrize(
-    ["regularization", "expected"], [("l0", 6), ("l1", 20), ("l2", 76)]
+    ["regularization", "lam", "expected"],
+    [
+        ("l0", 2, 4),
+        ("l1", 2, 14),
+        ("l2", 2, 58),
+        ("weighted_l0", np.array([[3, 2]]).T, 5),
+        ("weighted_l1", np.array([[3, 2]]).T, 16),
+        ("weighted_l2", np.array([[3, 2]]).T, 62),
+    ],
 )
-def test_get_regularization_1d(regularization, expected):
-    data = np.array([[-2, 3, 5]]).T
-    lam = np.array([[2]])
+def test_get_regularization(regularization, lam, expected):
+    data = np.array([[-2, 5]]).T
 
     reg = get_regularization(regularization)
     result = reg(data, lam)
     assert result == expected
 
 
-@pytest.mark.parametrize(
-    ["regularization", "expected"], [("l0", 10), ("l1", 56), ("l2", 416)]
-)
-def test_get_regularization_2d(regularization, expected):
-    data = np.array([[-2, 3, 5], [7, 11, 0]]).T
-    lam = np.array([[2]])
-
+@pytest.mark.parametrize("regularization", ["l0", "l1", "l2"])
+@pytest.mark.parametrize("lam", [1, np.array([1]), np.array([[1]])])
+def test_get_prox_and_regularization_shape(regularization, lam):
+    data = np.array([[-2, 5]]).T
     reg = get_regularization(regularization)
-    result = reg(data, lam)
-    assert result == expected
+    reg_result = reg(data, lam)
+    prox = get_prox(regularization)
+    prox_result = prox(data, lam)
+    assert reg_result is not None
+    assert prox_result is not None
```

Comment:
What is the purpose of these tests?  By the static typing, `reg_result` and `prox_result` are never `None`

Reply:
This is to check if it passes the validation. Do you know a better way to test this?